File: boilerplate/java/order-service/src/main/java/com/example/orderservice/application/usecases/PlaceOrderUseCase.java
package com.example.orderservice.application.usecases;

import com.example.orderservice.application.dtos.CreateOrderCommand;
import com.example.orderservice.application.dtos.OrderResult;

import java.util.UUID;

public interface PlaceOrderUseCase {
    OrderResult execute(UUID customerId, CreateOrderCommand command);
}

File: boilerplate/python/order-service/src/application/usecases/place_order_use_case.py
"""Application use case interface.

Command type — orchestrates domain logic and returns a result.
"""

from abc import ABC, abstractmethod

from application.dtos import CreateOrderCommand, OrderResult


class PlaceOrderUseCase(ABC):
    """Application layer port for placing orders."""

    @abstractmethod
    def execute(self, command: CreateOrderCommand) -> OrderResult:
        """Execute the order placement.

        Args:
            command: Validated create-order command

        Returns:
            OrderResult with order_id, status, created_at
        """
        ...
